### PR TITLE
Feat: EVM gas cost optimizations

### PR DESCRIPTION
**Pull request by mrLSD**
## Description

Based on `SputnikVM` EVM optimizations: aurora-is-near/sputnikvm/pull/48

### Gas cost

➡️  A lot of tests show gas decreased to: `1-2%`
➡️  For some tests gas cost increased, it'is related to: aurora-is-near/sputnikvm/pull/46 as it's requires additional NEAR storage read operation.

### CODE REVIEW

File: engine-tests/src/utils/mod.rs [hunk 1]
```diff
@@ -1029,10 +1029,10 @@ pub fn assert_gas_bound(total_gas: u64, tgas_bound: u64) {
 
 /// Returns true if `abs(a - b) / max(a, b) <= x / 100`. The implementation is written differently than
 /// this simpler formula to avoid floating point arithmetic.
-pub const fn within_x_percent(x: u64, a: u64, b: u64) -> bool {
+#[allow(clippy::cast_precision_loss, clippy::as_conversions)]
+pub fn within_x_percent(x: u64, a: u64, b: u64) -> bool {
     let (larger, smaller) = if a < b { (b, a) } else { (a, b) };
-
-    (100 / x) * (larger - smaller) <= larger
+    (100. / x as f64) * (larger as f64 - smaller as f64) <= larger as f64
```
Comment (birchmd) [inline engine-tests/src/utils/mod.rs:1035]: Why the change to `f64`?

Comment (mrLSD) [reply engine-tests/src/utils/mod.rs:1035]: For `u64` example: `100/24 == 100/21`, it's too broad an assumption. ))

Comment (birchmd) [reply engine-tests/src/utils/mod.rs:1035]: Maybe it makes more sense to remove the division from the expression entirely and instead write the condition as

```
100 * (larger - smaller) <= larger * x
```

Comment (mrLSD) [reply engine-tests/src/utils/mod.rs]: Done: [ab48316](https://github.com/aurora-is-near/aurora-engine/pull/934/commits/ab483162d5434fd7a6647aa82639b31c120d0089)

File: engine/src/engine.rs [hunk 2]
```diff
@@ -1915,6 +1921,29 @@ impl<'env, I: IO + Copy, E: Env, M: ModExpAlgorithm> Backend for Engine<'env, I,
         result
     }
 
+    /// Check is storage empty for address
+    /// EIP-7610: non-empty storGE
```
Comment (aleksuss) [inline engine/src/engine.rs:1925]:
```suggestion
    /// EIP-7610: non-empty storage
```

Comment (mrLSD) [reply engine/src/engine.rs]: Done: b93ff81c906730be4bb75eb89c6b401ee0217933

**Review by birchmd — APPROVED** [review]

**Review by aleksuss — APPROVED** [review]
